core/src/main/java/com/amplifyframework/core/category/Category.java

```diff
@@ -184,6 +183,13 @@ public final P getPlugin(@NonNull final String pluginKey) throws IllegalStateExc
                 "Tried to get a plugin but that plugin was not present. " +
                 "Check if the plugin was added originally or perhaps was already removed."
             );
+        } else if (State.FAILED.equals(state.get())) {
+            throw new IllegalStateException(
+                "Failed to get plugin because either configuration or initialization failed.  Check for " +
+                "configuration failures by logging any exceptions thrown by Amplify.configure().  Check for " +
+                "initialization failures by subscribing to the InitializationStatus.FAILED event on the Hub channel " +
+                "for the respective category with Amplify.Hub.subscribe()"
+            );
```

Comment:
```suggestion
            throw new IllegalStateException(
                "Failed to get plugin because either configuration or initialization had previously failed. " +
                "Check for configuration failures by logging any exceptions thrown by Amplify.configure(). " +
                "Check for initialization failures by subscribing to the InitializationStatus.FAILED event on "
                " the Hub channel for the respective category with Amplify.Hub.subscribe()."
            );
```